esp-hal-common/src/system.rs

```diff
@@ -56,7 +63,50 @@ pub enum Peripheral {
     #[cfg(esp32c6)]
     Twai1,
 }
+#[cfg(not(any(esp32, esp32c6)))]
```

Comment:
Software CPU interrupts are supported on all chips, so I don't think we need the feature gates? It might the case that they are not included in the PACs.

Reply:
At least we use SW_INTR3 in esp-wifi

Reply:
It's in the INTPRI peripheral in the C6 hal. I don't have one for testing however, so i don't feel confident implementing it.

Reply:
For the esp32 it's in the DPORT, so i guess support there is just a matter of enabling it